lblod/frontend-loket
# [DL-5866] Toezicht: Add a `DecisionArticles` field and validator

This adds a new custom form field that can be used to refer to multiple worship decisions.

Depends on https://github.com/lblod/app-digitaal-loket/pull/575

## To test:

1. ~~Run both this and the backend PR~~ The backend PR now includes the feature image of this PR.
2. ~~Apply the following patch to the active form file (check ACTIVE_FORM_FILE in docker-compose.yml)~~ Should no longer be needed with the latest backend commits.

<details>
<summary>Patch</summary>

```diff
diff --git a/config/semantic-forms/20240507113456-forms.ttl b/config/semantic-forms/20240507113456-forms.ttl
index e90b4334..7ab5a0f2 100644
--- a/config/semantic-forms/20240507113456-forms.ttl
+++ b/config/semantic-forms/20240507113456-forms.ttl
@@ -1737,6 +1737,42 @@ fields:eb089ca1-2c5b-42a4-a781-f009c46ac583 a form:Field ;
     form:displayType displayTypes:conceptSchemeSelector ;
     sh:group fields:aDynamicPropertyGroup .
 
+##########################################################
+# Worship Decisions cross reference selector
+##########################################################
+fields:788b9a98-684e-4d20-ae22-0ae610e6b494 a form:Field ;
+    mu:uuid "788b9a98-684e-4d20-ae22-0ae610e6b494" ;
+    sh:name "Artikels" ;
+    sh:order 102 ;
+    sh:path eli:has_part ;
+    form:validations
+      [ a form:RequiredConstraint ;
+        form:grouping form:Bag ;
+        sh:path eli:is_about ;
+        sh:path eli:has_part ;
+        sh:resultMessage "Dit veld is verplicht."@nl
+      ] ;
+    form:displayType displayTypes:worshipDecisionsCrossReferenceSelector ;
+    sh:group fields:aDynamicPropertyGroup .
+
+
+fields:type a form:Field ;
+    sh:name "Hidden field" ;
+    sh:order 102 ;
+    sh:path (<http://data.europa.eu/eli/ontology#has_part> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type>);
+    sh:group fields:aDynamicPropertyGroup .
+
+fields:type_doc a form:Field ;
+    sh:name "Hidden field" ;
+    sh:order 1001 ;
+    sh:path (<http://data.europa.eu/eli/ontology#has_part> <http://data.europa.eu/eli/ontology#type_document>);
+    sh:group fields:aDynamicPropertyGroup .
+
+fields:refers_to a form:Field ;
+    sh:name "Hidden field" ;
+    sh:order 1001 ;
+    sh:path (<http://data.europa.eu/eli/ontology#has_part> <http://data.europa.eu/eli/ontology#refers_to>);
+    sh:group fields:aDynamicPropertyGroup .
 
 
 ##########################################################
@@ -2005,6 +2041,9 @@ fieldGroups:91045004-2066-40f3-aac1-bf2309eb2b99 a form:FieldGroup ;
                       
                       ###Eredienst selector###
                       fields:00a7bce6-2925-4794-93cb-ebd571b28831,
+
+                      ###Worship Decisions Cross Reference selector
+                      fields:788b9a98-684e-4d20-ae22-0ae610e6b494,
               
                       ###Rapportjaar###
                       fields:41737f90-02d6-4036-8d60-5d5b6ccf939c,
@@ -2655,6 +2694,9 @@ fieldGroups:09fd371f-2afe-4b14-81a9-e780876de077 a form:FieldGroup ;
                       ###Betreffend-bestuur-van-de-eredienst-(eredienst-selector)###
                       fields:00a7bce6-2925-4794-93cb-ebd571b28831,
 
+                      ###Worship Decisions Cross Reference selector
+                      fields:788b9a98-684e-4d20-ae22-0ae610e6b494,
+
                       ###Welk-beslissingsorgaan-nam-het-besluit?###
                       fields:4c7820f0-4011-4ab4-a16a-e128800e11bc,
 
@@ -2774,6 +2816,9 @@ fieldGroups:ae168679-82cf-4df6-8803-cbcbc6610f47 a form:FieldGroup ;
                       ###Eredienst selector###
                       fields:00a7bce6-2925-4794-93cb-ebd571b28831,
 
+                      ###Worship Decisions Cross Reference selector
+                      fields:788b9a98-684e-4d20-ae22-0ae610e6b494,
+
                       ###Gaat-het-over-het-origineel-document-of-over-een-wijziging?###
                       fields:a8f6a6cb-dbb8-488c-878d-05603791a9eb,
 
@@ -6255,6 +6300,9 @@ fieldGroups:abb02ef6-581d-4876-aa45-acd1fbc4f49b a form:FieldGroup ;
                       ###Eredienst selector###
                       fields:eb089ca1-2c5b-42a4-a781-f009c46ac583,
 
+                      ###Worship Decisions Cross Reference selector
+                      fields:788b9a98-684e-4d20-ae22-0ae610e6b494,
+
                       ###Datum-zitting/besluit###
                       fields:3dd6ed93-40f7-4d70-a6cb-f4de53dc8bfb,
 
@@ -6295,6 +6343,9 @@ fieldGroups:253be7ce-f723-4e82-8b24-3e55e0602aaa a form:FieldGroup ;
                       ###Eredienst selector###
                       fields:eb089ca1-2c5b-42a4-a781-f009c46ac583,
 
+                      ###Worship Decisions Cross Reference selector
+                      fields:788b9a98-684e-4d20-ae22-0ae610e6b494,
+
                       ###Datum-zitting/besluit###
                       fields:3dd6ed93-40f7-4d70-a6cb-f4de53dc8bfb,

```
</details>

3. Follow the test setup instructions in the backend PR: https://github.com/lblod/app-digitaal-loket/pull/575
4. Create a new decision as Gemeente Aalst and test the component.
